diff -ruN a/gmloader/texture_arm32.cpp b/gmloader/texture_arm32.cpp
--- a/gmloader/texture_arm32.cpp	2026-10-03 16:06:14.810073850 -0400
+++ b/gmloader/texture_arm32.cpp	2026-10-05 19:55:56.789649989 -0400
@@ -46,12 +46,12 @@
 		for (;;) {
 			if (*p == 0xE5900020) { // LDR R0, [R0,#0x20]
 				warning("Patching LoadTextureFromPNG to variant #4\n");
-				hook_address(mod, (uintptr_t)&LoadTextureFromPNG, (uintptr_t)&LoadTextureFromPNG_4);
+				hook_address(mod, (uintptr_t)LoadTextureFromPNG, (uintptr_t)&LoadTextureFromPNG_4);
 				found = 1;
 				break;
 			} else if (*p == 0xE5900024) { // LDR R0, [R0,#0x24]
 				warning("Patching LoadTextureFromPNG to variant #3\n");
-				hook_address(mod, (uintptr_t)&LoadTextureFromPNG, (uintptr_t)&LoadTextureFromPNG_3);
+				hook_address(mod, (uintptr_t)LoadTextureFromPNG, (uintptr_t)&LoadTextureFromPNG_3);
 				found = 1;
 				break;
 			}
@@ -66,11 +66,11 @@
 		switch (*LoadTextureFromPNG >> 16) {
 		case 0xE92D:
 			warning("Patching LoadTextureFromPNG to variant #1\n");
-			hook_address(mod, (uintptr_t)&LoadTextureFromPNG, (uintptr_t)&LoadTextureFromPNG_1);
+			hook_address(mod, (uintptr_t)LoadTextureFromPNG, (uintptr_t)&LoadTextureFromPNG_1);
 			break;
 		case 0xE590:
 			warning("Patching LoadTextureFromPNG to variant #2\n");
-			hook_address(mod, (uintptr_t)&LoadTextureFromPNG, (uintptr_t)&LoadTextureFromPNG_2);
+			hook_address(mod, (uintptr_t)LoadTextureFromPNG, (uintptr_t)&LoadTextureFromPNG_2);
 			break;
 		default:
 			fatal_error("Error: Unrecognized LoadTextureFromPNG signature: 0x%08X.", *LoadTextureFromPNG);
